### DISCUSSION TITLE
Kafka Exporter Crashing

### DISCUSSION THREAD

**Discussion by saloni-2404** [strimzi/strimzi-kafka-operator#5244] [General]
Hi Jakub,
We have a kafka-exporter pod crashing continuously, we tried the following: decreased the load of metrics being pulled from the exporter, still it is crashing.
Background:
Earlier we saw errors such as "kafka broker not online"
Now, we no longer see that, but observe healthiness and liveliness probe failing, and the pod continuously crashing.
Attaching the logs and describe.

**Comment by saloni-2404** [comment]
[exporter-desc.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6753123/exporter-desc.txt)
[exporter-logs.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6753124/exporter-logs.txt)
[exporter-logs-prev.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6753125/exporter-logs-prev.txt)
[op-logs.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6753126/op-logs.txt)

**Comment by scholzj** [comment]
What version of Strimzi are you using? The 0.24.0 release has some fixes for the Kafka Exporter.

**Reply by saloni-2404** [reply]
KAFKA_EXPORTER_KAFKA_VERSION:  2.4.0
Strimzi operator: 0.17.0

**Reply by saloni-2404** [reply]
Are there any suggestions for 0.17.0, please?

**Reply by scholzj** [reply]
Sorry, I do not have any. I do not know what the problem is.

**Reply by saloni-2404** [reply]
Hey Jakub, in the operator logs we noticed:

2021-07-06 05:32:39 WARN  AbstractOperator:193 - Reconciliation #34963(timer) Kafka(sedsep-dev/kafka-opstate): Failed to acquire lock lock::sedsep-dev::Kafka::kafka-opstate within 10000ms.

Is there any reason for this error, could this be why the exporter is crashing?

**Reply by scholzj** [reply]
It is not an error and it has nothing to do with the Kafka Exporter issues. In the docs, there is an FAQ section explaining this message. As I said before, the Kafka Exporter is buggy - some / most of these should be fixed in 0.24.

**Reply by saloni-2404** [reply]
Ok sure, thanks Jakub!
